metadatastore/doc.py

```diff
@@ -0,0 +1,229 @@
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+import six
+import collections
+from functools import reduce
+
+_HTML_TEMPLATE = """
+<table>
+{% for key, value in document | dictsort recursive %}
+  <tr>
+    <th> {{ key }} </th>
+    <td>
+      {% if value.items %}
+        <table>
+          {{ loop(value | dictsort) }}
+        </table>
+        {% else %}
+          {% if key == 'time' %}
+            {{ value | human_time }}
+          {% else %}
+            {{ value }}
+          {% endif %}
+        {% endif %}
+    </td>
+  </tr>
+{% endfor %}
+</table>
+"""
+
+
+class DocumentIsReadOnly(Exception):
+    pass
+
+
+class Document(dict):
```

Comment:
Document used to be dottable. Can you implement the `__getitem__` function for it?